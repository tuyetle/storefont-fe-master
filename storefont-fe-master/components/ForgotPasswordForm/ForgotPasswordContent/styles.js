import { css } from 'glamor';
import themes from '~/styles/themes';
import variables from '~/styles/variables';

export default {
  title: css({
    ...themes['font-notoserif'],
    fontSize: '1.250em',
    '@media (min-width: 768px)': {
      fontSize: '1.563em',
    },
    textAlign: 'center',
  }),
  label: css({
    fontSize: '0.875em',
  }),
  description: css({
    marginTop: '1em',
    textAlign: 'center',
    marginBottom: '2em',
    fontSize: '0.938em',
  }),

  input: css({
    border: 'none',
    marginTop: '0.875em',
    marginBottom: '0.188em',
    borderRadius: '0',
    borderBottom: '1px solid lightgray',
    '&:focus': {
      borderBottomColor: variables.$focusBorderBlue,
      boxShadow: 'none',
      outline: 'none',
    },

    '&.form-control:focus': {
      boxShadow: 'none',
      border: '1px solid lightgray',
    },
  }),

  submitButton: css({
    ...themes['submit-button'],
  }),

  disableSubmitButton: css({
    ...themes['disable-submit-button'],
  }),
};
