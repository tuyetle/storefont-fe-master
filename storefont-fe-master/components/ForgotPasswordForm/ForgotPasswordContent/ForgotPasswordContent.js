import React, { Fragment } from 'react';
import PropTypes from 'prop-types';
import { noop, isEmpty } from 'lodash';
import { Field } from 'react-form';
import { translate } from 'react-i18next';
import EmailOrPhoneFormField from '~/components/common/Forms/EmailOrPhoneFormField/EmailOrPhoneFormField';
import styles from './styles';

const ForgotPasswordContent = (props) => {
  const { t, formApi } = props;
  return (
    <Fragment>
      <div className={`${styles.title}`} > {t('forgotPassword')} </div>
      <form onSubmit={formApi.submitForm}>
        <div className={`${styles.description}`} > {t('forgotPasswordDescription')} </div>
        <div className="form-group">
          <strong className={styles.label}>{t('emailOrPhone')}</strong>
          <Field
            field="emailOrPhone"
            component={EmailOrPhoneFormField}
            className={`${styles.input} form-control`}
          />
        </div>
        <input
          className={isEmpty(formApi.values) ? `${styles.disableSubmitButton}` : `${styles.submitButton}`}
          type="submit"
          disabled={isEmpty(formApi.values)}
          value={t('sendNewPassword')}
        />
      </form>
    </Fragment>
  );
};
ForgotPasswordContent.propTypes = {
  formApi: PropTypes.shape({
    submitForm: PropTypes.func,
  }).isRequired,
  t: PropTypes.func,
};

ForgotPasswordContent.defaultProps = {
  t: noop,
};
export default translate(['auth'])(ForgotPasswordContent);
