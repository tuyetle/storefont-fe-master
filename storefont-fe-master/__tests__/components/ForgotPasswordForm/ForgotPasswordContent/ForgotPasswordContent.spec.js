import React from 'react';
import ForgotPasswordContent from '~/components/ForgotPasswordForm/ForgotPasswordContent/ForgotPasswordContent';
import renderer from 'react-test-renderer';

describe('(Component) ForgotPasswordContent', () => {
  let component;
  let props;
  let spyCallback;

  beforeEach(() => {
    props = {
      t(value) { return value; },
      formApi: { submitForm() {} },
    };
    spyCallback = jest.spyOn(props.formApi, 'submitForm');
    component = mount(<ForgotPasswordContent {...props} />);
  });

  it('should render conrectly', () => {
    const tree = renderer.create(component).toJSON();

    expect(tree).toMatchSnapshot();
  });
  it('should call submitForm when click submit', () => {
    const btn = component.find('input[type="submit"]');
    btn.simulate('submit');
    expect(spyCallback).toHaveBeenCalled();
  });
});
