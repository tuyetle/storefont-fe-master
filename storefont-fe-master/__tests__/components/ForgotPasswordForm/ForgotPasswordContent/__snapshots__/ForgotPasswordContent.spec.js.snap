// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`(Component) ForgotPasswordContent should render conrectly 1`] = `
Array [
  <div
    className="css-c2egwx"
  >
     
    forgotPassword
     
  </div>,
  <form
    onSubmit={[MockFunction]}
  >
    <div
      className="css-n6jj6g"
    >
       
      forgotPasswordDescription
       
    </div>
    <div
      className="form-group"
    >
      <strong
        className={
          Object {
            "data-css-q6gydh": "",
          }
        }
      >
        emailOrPhone
      </strong>
      Field
    </div>
    <input
      className="css-7qje71"
      disabled={true}
      type="submit"
      value="sendNewPassword"
    />
  </form>,
]
`;
